### ISSUE TITLE
Dependency node-webcrypto-ossl

### ISSUE THREAD

**Issue by datacyclist** [gausma/libsignal-service-javascript#3]
Any chance of an easy fix for the dependency node-webcrypto-ossl?

 "node-webcrypto-ossl": "^1.0.48",

That lib seems to be deprecated, so the https://github.com/gausma/nodered-contrib-signal-client no longer works.

**Comment by jwiesel** [comment]
I just updated a Node-Red instance and realized, that the dependency on proxy-agent which uses agent-base is also outdated.
Would be great to update that dependency, too. 
"proxy-agent": "^3.1.0", --> "proxy-agent": "^5.0.0",
Message in Node-Red

> Patched https.request function detected. This will break theHTTP Request node. The original code has now been restored.This is likely caused by a contrib node including an old version ofthe 'agent-base@<5.0.0' module.You can identify what node is at fault by running: npm list agent-base in your Node-RED user directory

Getting https://github.com/gausma/nodered-contrib-signal-client to work again would be a huge help!

**Comment by jwiesel** [comment]
P.S: The fork of Damme of this repo moves from node-webcrypto-ossl (deprecated) to @peculiar/webcrypto
https://github.com/Damme/libsignal-service-javascript/commit/cde16fb18738425e63c8a54b1fb1a0523c1e8eaa

**Comment by skylord123** [comment]
> I just updated a Node-Red instance and realized, that the dependency on proxy-agent which uses agent-base is also outdated. Would be great to update that dependency, too. "proxy-agent": "^3.1.0", --> "proxy-agent": "^5.0.0", Message in Node-Red
> 
> > Patched https.request function detected. This will break theHTTP Request node. The original code has now been restored.This is likely caused by a contrib node including an old version ofthe 'agent-base@<5.0.0' module.You can identify what node is at fault by running: npm list agent-base in your Node-RED user directory
> 
> Getting https://github.com/gausma/nodered-contrib-signal-client to work again would be a huge help!

Yeah, `proxy-agent` needs to be updated. I am the developer of `node-red-contrib-matrix-chat` and found some of my users were getting errors with HTTP requests. I tracked it down to them using `node-red-contrib-signal-client` which then uses this module.

When they uninstalled `node-red-contrib-signal-client` their http requests started working again. `proxy-agent` needs to be updated to a version above `5.0.0` to fix this.

I opened [an issue](https://github.com/gausma/nodered-contrib-signal-client/issues/36) on `node-red-contrib-signal-client` as well but the problem really lies here.

**Comment by gruijter** [comment]
Was this resolved? Having issues with node-red-contrib-signal-client

**Comment by rgstephens** [comment]
No, this is still an open issue from what I can see.

**Comment by skylord123** [comment]
@gausma have you abandoned this fork?
